src/brain/gocdSlackFeeds/index.test.ts

```diff
@@ -1745,6 +1746,199 @@ Please do not ignore this message just because the environment is not SaaS, beca
+    await handler(gocdPayloadSuccess);
+    expect(bolt.client.chat.postMessage).toHaveBeenCalledTimes(3);
+    const postCalls = bolt.client.chat.postMessage.mock.calls;
+    postCalls.sort(sortMessages);
+
+    expect(postCalls[0][0]).toMatchObject({
+      channel: 'channel_id',
+      text: '',
+      blocks: [
+        slackblocks.header(
+          slackblocks.plaintext('run-custom-job stage update')
+        ),
+        {
+          elements: [
+            slackblocks.markdown('❌ *checks*'),
+            slackblocks.markdown(
+              '<https://deploy.getsentry.net/go/pipelines/run-custom-job/20/checks/1|Failed>'
+            ),
+          ],
+        },
+        slackblocks.divider(),
+        slackblocks.context(
+          slackblocks.markdown(
+            `cc'ing the user who started this deployment: <@GoCD_Slack_User>`
+          )
+        ),
+      ],
+    });
+    expect(postCalls[1][0]).toMatchObject({
+      text: 'GoCD deployment started by <@GoCD_Slack_User>',
+      channel: FEED_GOCD_JOB_RUNNER_CHANNEL_ID,
+      attachments: [
+        {
+          color: Color.DANGER,
+          blocks: [
+            slackblocks.section(
+              slackblocks.markdown('*sentryio/run-custom-job*')
+            ),
+            {
+              elements: [
+                slackblocks.markdown('Deploying'),
+                slackblocks.markdown(
+                  '<https://github.com/getsentry/getsentry/commits/2b0034becc4ab26b985f4c1a08ab068f153c274c|getsentry@2b0034becc4a>'
+                ),
+              ],
+            },
+            slackblocks.divider(),
+            {
+              elements: [
+                slackblocks.markdown('❌ *checks*'),
+                slackblocks.markdown(
+                  '<https://deploy.getsentry.net/go/pipelines/run-custom-job/20/checks/1|Failed>'
+                ),
+              ],
+            },
+          ],
+        },
+      ],
+    });
+    expect(postCalls[2][0]).toMatchObject({
+      text: 'GoCD deployment started by <@GoCD_Slack_User>',
+      channel: FEED_DEPLOY_CHANNEL_ID,
+      attachments: [
+        {
+          color: Color.DANGER,
+          blocks: [
+            slackblocks.section(
+              slackblocks.markdown('*sentryio/run-custom-job*')
+            ),
+            {
+              elements: [
+                slackblocks.markdown('Deploying'),
+                slackblocks.markdown(
+                  '<https://github.com/getsentry/getsentry/commits/2b0034becc4ab26b985f4c1a08ab068f153c274c|getsentry@2b0034becc4a>'
+                ),
+              ],
+            },
+            slackblocks.divider(),
+            {
+              elements: [
+                slackblocks.markdown('❌ *checks*'),
+                slackblocks.markdown(
+                  '<https://deploy.getsentry.net/go/pipelines/run-custom-job/20/checks/1|Failed>'
+                ),
+              ],
+            },
+          ],
+        },
+      ],
+    });
+  });
+
```

Comment:
What are your thoughts on adding one more test for when the result is not unknown but there is no user?

Reply:
Sure, that's a good idea. Adding now.